### PR TITLE
tests: use unique namespace to prevent collisions

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan-src/pull/1115

### CODE REVIEW

File: tests/PHPStan/Rules/PhpDoc/IncompatiblePhpDocTypeRuleTest.php [hunk 1]
```diff
@@ -210,8 +210,6 @@ public function testValueOfEnum(): void
 			$this->markTestSkipped('This test needs PHP 8.1');
 		}
 
-		require __DIR__ . '/data/value-of-enum.php';
```
Comment (staabm) [inline tests/PHPStan/Rules/PhpDoc/IncompatiblePhpDocTypeRuleTest.php:213]: also drop this unnecessary require

File: tests/PHPStan/Rules/PhpDoc/data/value-of-enum.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace ValueOfEnum;
+namespace ValueOfEnumPhpdoc;
```
Comment (staabm) [inline tests/PHPStan/Rules/PhpDoc/data/value-of-enum.php:3]: this namespace collided with the one in `tests/PHPStan/Analyser/data/value-of-enum.php` and therefore we get same-class errors when running tests in a single process, see https://github.com/phpstan/phpstan-src/pull/1115#issuecomment-1078779940

**Comment by Slamdunk** [comment]
As mentioned https://github.com/phpstan/phpstan-src/pull/1115#issuecomment-1078779940 this is still not enough.
The default ParaTest behavior is to spawn a new process for each test, and that hides eventual collisions.
The current patch proposed here still raises a fatal error when changes introduced in https://github.com/phpstan/phpstan-src/pull/1082 are run together in PHPUnit:

```
$ vendor/bin/phpunit --filter '/(value-of-enum|testValueOfEnum)/' --debug
PHP Fatal error:  Cannot declare class ValueOfEnumPhpdoc\Country, because the name is already in use in ./phpstan-src/tests/PHPStan/Rules/PhpDoc/data/value-of-enum.php on line 7
```

**Comment by staabm** [comment]
Its an improvement nevertheless, even if its not yet enough for the paratest case

**Comment by Slamdunk** [comment]
Sure, I just wanted to make everyone aware ^^

**Comment by ondrejmirtes** [comment]
Please fix the build failure.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**